File: Leetcode/167 - Two Sum II - Input Array Is Sorted/solution.ts
function twoSum(numbers: number[], target: number): number[] {
    let low: number = 0, high: number = numbers.length - 1;
    while (low < high) {
        const sum: number = numbers[low] + numbers[high];
        if (sum === target) return [low + 1, high + 1];
        else if (sum > target) high--;
        else low++;
    }
    return [];
};

File: Leetcode/167 - Two Sum II - Input Array Is Sorted/solution.py
def twoSum(self, numbers: List[int], target: int) -> List[int]:
    low, high = 0, len(numbers) - 1
    while low < high:
        s = numbers[low] + numbers[high]
        if s == target:
            return [low + 1, high + 1]
        elif s > target:
            high -= 1
        else:
            low += 1
    return []